client/src/components/AddMates/AddMatesPage.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Button,
+  Col,
+  DatePicker,
+  Divider,
+  Form,
+  Input,
+  Layout,
+  Row,
+  Space,
+  Typography,
+} from "antd";
+import React, { FC, useState } from "react";
+import { Prompt } from "react-router";
+
+import { AddMateFormType, AddMateType } from "../types";
+import AddMateCard from "./AddMateCard";
+
+const { Title } = Typography;
+const { Footer } = Layout;
+
+const AddMatesPage: FC = () => {
+  const [form] = Form.useForm<AddMateFormType>();
+  const [mates, setMates] = useState<AddMateType[]>([]);
+
+  const onAddMate = async ({ name, lastSeen }: AddMateFormType) => {
+    setMates((prevArray) => [
+      ...prevArray,
+      { name, lastSynced: lastSeen.toDate(), ts: new Date().getTime() },
+    ]);
+    form.resetFields();
+  };
+
+  const removeMate = (mate: AddMateType) => {
+    setMates((mateArr) =>
+      mateArr.filter(
```

Comment:
might be easier to assign unique id when you create the m8, so you can just filter for that instead of doing full equality check

Reply:
changed to incremental id